client/src/app.js

```diff
@@ -1592,3 +1594,43 @@ function initgenome(g) {
 	}
 	return null
 }
+
+async function launchDisco(arg, app) {
+	if (!arg.genome) throw '"genome" parameter missing'
+	const genomeObj = app.genomes[arg.genome]
+	if (!genomeObj) throw 'unknown genome'
+	if (!Array.isArray(arg.disco.mlst)) throw 'arg.disco.mlst[] not array'
+	const opts = {
+		holder: app.holder0,
+		vocabApi: {
+			// api is required by plot.app.js, so create a mock one for the adhoc data
+			vocab: { terms: [] },
+			main: () => {},
+			getTermdbConfig: () => {
+				return {}
+			}
+		},
+		state: {
+			args: {
+				data: arg.disco.mlst,
+				genome: genomeObj
+			},
+			plots: [
+				{
+					chartType: 'Disco',
+					subfolder: 'disco',
+					extension: 'ts'
+					/*
+					overrides: {
+						label: {
+							showPrioritizeGeneLabelsByGeneSets: !!genomeObj.geneset
+						}
+					}
+					*/
+				}
+			]
+		}
+	}
+	const plot = await import('#plots/plot.app.js')
+	const plotAppApi = await plot.appInit(opts)
```

Comment:
@xzhou82 should `return plotAppApi`, so that it can return an app with a dispatch method that can be used by the embedder to update the plot - to make it interactive.
```js
const disco = runproteinpaint({..., launchDisco: ...})
```